src/c++/perf_analyzer/genai-pa/llm_inputs/llm_inputs.py

```diff
@@ -0,0 +1,282 @@
+        SYSTEM_ROLE_LIST = ["system_prompt"]
+        USER_ROLE_LIST = ["question", "article"]
+
+        system_role_headers, user_role_headers = [], []
+        if "features" in dataset_json.keys():
+            for index, feature in enumerate(dataset_json["features"]):
+                if feature["name"] in SYSTEM_ROLE_LIST:
+                    system_role_headers.append(feature["name"])
+                if feature["name"] in USER_ROLE_LIST:
+                    user_role_headers.append(feature["name"])
+
+        assert system_role_headers is not None or user_role_headers is not None
+
+        return system_role_headers, user_role_headers
+
+    @classmethod
+    def _populate_pa_json(
+        cls,
+        dataset_json: Dict,
+        system_role_headers: List[str],
+        user_role_headers: List[str],
+        model_name: str,
+        add_streaming: bool,
+    ) -> Dict:
+        pa_json = LlmInputs._create_empty_pa_json()
+
+        for entry in dataset_json["rows"]:
+            pa_json["data"][0]["payload"].append({"messages": []})
+
+            for header in entry["row"]:
+                new_message = LlmInputs._create_new_message(
+                    header, system_role_headers, user_role_headers, entry["row"][header]
+                )
+
+                pa_json = LlmInputs._add_new_message_to_json(
+                    pa_json, entry["row_idx"], new_message
+                )
+
+            pa_json = LlmInputs._add_optional_tags_to_json(
+                pa_json, entry["row_idx"], model_name, add_streaming
+            )
+
+        return pa_json
+
+    @classmethod
+    def _create_empty_pa_json(cls) -> Dict:
+        empty_pa_json = deepcopy(LlmInputs.EMPTY_JSON_IN_PA_FORMAT)
+
+        return empty_pa_json
+
+    @classmethod
+    def _create_new_message(
+        cls,
+        header: str,
+        system_role_headers: List[str],
+        user_role_headers: List[str],
+        content: str,
+    ) -> Dict:
+        if header in system_role_headers:
+            new_message = {
+                "role": "system",
+                "content": content,
+            }
+        elif header in user_role_headers:
+            new_message = {
+                "role": "user",
+                "content": content,
+            }
+        else:
+            new_message = {}
+
+        return new_message
+
+    @classmethod
+    def _add_new_message_to_json(
+        cls, pa_json: Dict, index: int, new_message: Dict
+    ) -> Dict:
+        if new_message:
+            pa_json["data"][0]["payload"][index]["messages"].append(new_message)
+
+        return pa_json
+
+    @classmethod
+    def _add_optional_tags_to_json(
+        cls, pa_json: Dict, index: int, model_name: str, add_streaming: bool
+    ) -> Dict:
+        if model_name:
+            pa_json["data"][0]["payload"][index]["model"] = model_name
+        if add_streaming:
+            pa_json["data"][0]["payload"][index]["streaming"] = "true"
```

Comment:
I told you wrong. It is supposed to be "stream", not "streaming"
https://platform.openai.com/docs/api-reference/chat/create#chat-create-stream

Reply:
Fixed